bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/async/AbstractRetryingOperation.java

```diff
@@ -276,29 +284,75 @@ protected void setException(Exception exception) {
    */
   protected abstract boolean onOK(Metadata trailers);
 
-  protected long getNextBackoff() {
+  protected Long getNextBackoff() {
     if (currentBackoff == null) {
-      currentBackoff = createBackoff();
+      // Historically, the client waited for "total timeout" after the first failure.  For now,
+      // that behavior is preserved, even though that's not the ideal.
+      //
+      // TODO: Think through retries, and create policy that works with the mental model most
+      //       users would have of relating to retries.  That would likely involve updating some
+      //       default settings in addition to changing the algorithm.
+      currentBackoff = exponentialRetryAlgorithm.createFirstAttempt();
     }
-    try {
-      return currentBackoff.nextBackOffMillis();
-    } catch (IOException e) {
-      return BackOff.STOP;
+    currentBackoff = exponentialRetryAlgorithm.createNextAttempt(currentBackoff);
+    if (!exponentialRetryAlgorithm.shouldRetry(currentBackoff)) {
+      return null;
+    } else {
+      return currentBackoff.getRetryDelay().toMillis();
     }
   }
 
+  @VisibleForTesting
+  protected boolean inRetryMode() {
+    return currentBackoff != null;
+  }
+
+  /**
+   * Either a response was found, or a timeout event occurred. Reset the information relating to
+   * Status oriented exception handling.
+   */
+  protected void resetStatusBasedBackoff() {
+    this.currentBackoff = null;
+    this.failedCount = 0;
+  }
+
   /**
    * <p>createBackoff.</p>
    *
-   * @return a {@link com.google.api.client.util.ExponentialBackOff} object.
+   * @return a {@link ExponentialRetryAlgorithm} object.
    */
+  private ExponentialRetryAlgorithm createRetryAlgorithm() {
+    int timeoutMs = retryOptions.getMaxElapsedBackoffMillis();
+
+    RetrySettings retrySettings = RetrySettings.newBuilder()
+
+        .setJittered(true)
+
+        // How long should the sleep be between RPC failure and the next RPC retry?
+        .setInitialRetryDelay(toDuration(retryOptions.getInitialBackoffMillis()))
+
+        // How fast should the retry delay increase?
+        .setRetryDelayMultiplier(retryOptions.getBackoffMultiplier())
+
+        // What is the maximum amount of sleep time between retries?
+        // There needs to be some sane number for max retry delay, and it's unclear what that
+        // number ought to be.  20% of timeout time was chosen because some number is needed
+        .setMaxRetryDelay(toDuration(timeoutMs / 5))
```

Comment:
I think it would be better to hardcode this to 1 minute

Reply:
I think that test should be tweaked

Reply:
What is a reasonable max and min, do you think?